armi/reactor/blocks.py

```diff
@@ -2027,6 +2027,11 @@ def rotate(self, rad):
         self._rotatePins(rotNum)
         self._rotateBoundaryParameters(rotNum)
         self._rotateDisplacement(rad)
+        self._rotatePinParameters(rotNum)
+
+    def _getParamsWhere(self, f: Callable[[parameters.Parameter], bool]):
```

Comment:
Two questions:

1. Do you want this here, or could this be a public method on `Composite` on `Parameter`?
    - I just notice there is already `Composite.getParamValuesAtZ()`.
    - I also notice that there is already a `Parameter.atLocation()`.
    - My THOUGHT is this looks generally useful, beyond `Block`s.
    - I'm okay being wrong, just wondering.
2. Should we have a unit test or two for this new method?

Reply:
All very good thoughts. I slapped this here because I needed it on `HexBlock` and couldn't find a good spot elsewhere.

It might make the most sense to be placed onto the `ParameterCollection` so it's auto assigned to everything that has parameters. And that makes sense: the thing that collects parameters allows you to do iteration over them.

I'll pull something into a separate PR because know I know we have a need for it, let's pull the discussion there, and then more "users who need things rotated" discussions can happen in this PR.

And yes, I will add tests now that it's being escalated to a public thing

Cool?

Reply:
That sounds great, man. I see [your ticket](https://github.com/terrapower/armi/issues/1898). Thanks for self-assigning it.